golang/outbound.go

```diff
@@ -56,12 +56,17 @@ func (c *Connection) beginCall(ctx context.Context, serviceName string) (*Outbou
 		return nil, err
 	}
 
+	headers := callHeaders{
+		CallerName: c.localPeerInfo.ProcessName,
```

Comment:
Once you do the hyperbahn integration, the channel itself will have a service name associated with it that it has advertised to hyperbahn.  That's the value that should go in here.  So it wouldn't come from the caller directly, it'd come from the channel (and is probably an option passed in to the channel - I'd look at the node implementation to see where they pass in the service name to advertise).  Note that this service name is an external route; you can actually have multiple thrift services registered under the same advertised service name (the thrift service that is being targeted gets passed as part of (I think) arg1).

I'm not sure how we are handling pure clients TBH.  These probably still have service names, even though they don't expose any services.  @Raynos?